<a href="https://colab.research.google.com/github/farankhandev/flyrank-ml-assignment-2/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/farankhandev/flyrank-ml-assignment-2/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


In [ ]:
import os
print(os.getcwd())

/content/flyrank-ml-assignment-2


In [ ]:
!git clone https://github.com/farankhandev/flyrank-ml-assignment-2.git

Cloning into 'flyrank-ml-assignment-2'...
remote: Enumerating objects: 111, done.
remote: Counting objects: 100% (111/111), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 111 (delta 28), reused 97 (delta 27), pack-reused 0 (from 0)
Receiving objects: 100% (111/111), 1.82 MiB | 10.51 MiB/s, done.
Resolving deltas: 100% (28/28), done.


In [ ]:
%cd flyrank-ml-assignment-2
!find . -name "*.csv"


/content/flyrank-ml-assignment-2/flyrank-ml-assignment-2
./outputs/refresh_queue_sample.csv
./data/raw/content_refresh_anonymized.csv


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Show 3 real numbers
print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))
print("\nMissing values per column:")
print(df.isnull().sum())

Number of rows: 30000
Number of columns: 44

Missing values per column:
content_id                    0
client_id                     0
search_volume              2468
competition                2468
competition_level          2610
cpc                        2468
content_type                  0
main_intent                2374
word_count                 7699
char_count                 7699
provider_used             21438
model_used                 5733
impressions_90d               0
clicks_90d                    0
pageviews_90d                 0
sessions_90d                  0
users_90d                     0
engaged_sessions_90d          0
ai_sessions_90d               0
scroll_events_90d             0
days_with_impressions         0
days_with_sessions            0
impressions_last_30d          0
clicks_last_30d               0
sessions_last_30d             0
impressions_prev_30d          0
clicks_prev_30d               0
sessions_prev_30d             0
content_age_days              0


### Quick look at the data

The starter dataset contains **30,000 rows** and **44 columns**, providing a substantial amount of information for analysis. I also observed that some fields contain missing values, for example **word_count has 7,699 missing values**, **provider_used has 21,438 missing values**, and **trend_pct has 3,388 missing values**. These statistics show that the dataset is large enough to support the Content Optimization lane while also requiring careful data cleaning before building any models.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## 4. Careful words: what I can and can't claim

This project uses observed search performance data to identify patterns that may help prioritize pages for content optimization. The results are intended to support decision-making and do not prove that one factor directly causes better search performance. This project does not predict Google's ranking algorithm or guarantee future search rankings.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.